File: Questão15.go
package main

import "fmt"


type Animal interface {
    EmitirSom() string
}


type Cachorro struct{}


func (Cachorro) EmitirSom() string {
    return "Au Au!"
}


type Gato struct{}


func (Gato) EmitirSom() string {
    return "Miau!"
}


func FazerSom(animal Animal) {
    fmt.Println(animal.EmitirSom())
}

func main() {
    var cachorro Animal = Cachorro{}
    var gato Animal = Gato{}

    FazerSom(cachorro) 
    FazerSom(gato)     }

File: Questão15.java
interface Animal {
    String emitirSom();
}

class Cachorro implements Animal {
    public String emitirSom() {
        return "Au Au!";
    }
}

class Gato implements Animal {
    public String emitirSom() {
        return "Miau!";
    }
}

public class Main {
    public static void fazerSom(Animal animal) {
        System.out.println(animal.emitirSom());
    }

    public static void main(String[] args) {
        Animal cachorro = new Cachorro();
        Animal gato = new Gato();

        fazerSom(cachorro); 
        fazerSom(gato);     
    }
}
